flashlib.core.robot/src/main/java/com/flash3388/flashlib/robot/systems/drive/PidSwerveWheel.java

```diff
@@ -0,0 +1,71 @@
+package com.flash3388.flashlib.robot.systems.drive;
+
+import java.util.function.DoubleSupplier;
+
+import com.flash3388.flashlib.io.devices.SpeedController;
+import com.flash3388.flashlib.robot.control.PidController;
+import com.jmath.vectors.Vector2;
+
+public class PidSwerveWheel implements SwerveWheel {
+
+    private final PidController mPidController;
+    private final SpeedController mForwardController;
+    private final SpeedController mRotationController;
+    private final DoubleSupplier mAngleDegreeSupplier;
+
+    public PidSwerveWheel(PidController pidController, SpeedController forwardController,
+            SpeedController rotationController, DoubleSupplier angleDegreeSupplier) {
+        mPidController = pidController;
+        mForwardController = forwardController;
+        mRotationController = rotationController;
+        mAngleDegreeSupplier = angleDegreeSupplier;
+
+        resetPid();
+    }
+
+    public PidSwerveWheel(SpeedController forwardController, SpeedController rotationController,
+            DoubleSupplier angleDegreeSupplier, double kp, double ki, double kd, double kf) {
+        this(new PidController(kp, kd, kd, kf), forwardController, rotationController, angleDegreeSupplier);
+    }
+
+    //this is bad, but I assume there is something built in, so I will wait for the PR on this one
+    public static double convertAngle(double angle) {
```

Comment:
To keep the degrees 0->360?
`Mathf.translateAngle`

Reply:
cool